Khan/eslint-plugin-khan
# Add a rule to disallow specific imports if flow isn't being used

The new `imports-requiring-flow` rule allows you to specify a which imports (either npm packages or local files) will require using `// @flow` in the file doing the import/require.  In order for this to work, you have to specify a `rootDir` in the rule configuration.

Test Plan:
- `yarn link`
- in devtools/khan-linter: `yarn link "eslint-plugin-khan"
- add an eslint.custom.js config to webapp containing the code listed below
- modify .eslintrc in webapp to include eslint.custom.js
- check that `404-page.jsx` and `foo` can't be imported

```js
// This has to be a .js file since we need to use __dirname
module.exports = {
    plugins: ["khan"],
    rules: {
        "khan/imports-requiring-flow": [
            "error",
            {
                rootDir: __dirname,
                modules: ["foo", "javascript/404-package/404-page.jsx"],
            },
        ],
    },
};
```

<img width="888" alt="Screen Shot 2019-10-23 at 12 04 31 PM" src="https://user-images.githubusercontent.com/1044413/67413764-cbcf4e00-f58f-11e9-9876-961354269fdc.png">
<img width="772" alt="Screen Shot 2019-10-22 at 5 03 41 PM" src="https://user-images.githubusercontent.com/1044413/67413791-d4c01f80-f58f-11e9-952c-2d01771c7086.png">
